lib/pdk/validators/metadata/task_metadata_lint.rb

```diff
@@ -0,0 +1,128 @@
+require 'pdk'
+require 'pdk/cli/exec'
+require 'pdk/validators/base_validator'
+require 'pdk/util'
+require 'pathname'
+require 'json-schema'
+require 'addressable'
+
+module PDK
+  module Validate
+    class TaskMetadataLint < BaseValidator
+      def self.name
+        'task-metadata-lint'
+      end
+
+      def self.pattern
+        'tasks/**.json'
+      end
+
+      def self.spinner_text(_targets = [])
+        _('Checking task metadata lint (%{targets}).') % {
+          targets: pattern,
+        }
+      end
+
+      def self.create_spinner(targets = [], options = {})
+        return if PDK.logger.debug?
+        options = options.merge(PDK::CLI::Util.spinner_opts_for_platform)
+
+        exec_group = options[:exec_group]
+        @spinner = if exec_group
+                     exec_group.add_spinner(spinner_text(targets), options)
+                   else
+                     TTY::Spinner.new("[:spinner] #{spinner_text(targets)}", options)
+                   end
+        @spinner.auto_spin
+      end
+
+      def self.stop_spinner(exit_code)
+        if exit_code.zero? && @spinner
+          @spinner.success
+        elsif @spinner
+          @spinner.error
+        end
+      end
+
+      def self.schema_file
+        schema = ''
+        if PDK::Util.package_install? && File.exist?(File.join(PDK::Util.package_cachedir, 'task.json'))
+          schema = File.read(File.join(PDK::Util.package_cachedir, 'task.json'))
+        else
+          PDK.logger.debug(_('Task Metadata Schema was not found in the cache. Now downloading from the forge.'))
+          begin
+            schema = open('https://forgeapi.puppet.com/schemas/task.json').read
+          rescue
+            raise PDK::CLI::FatalError, _('Unable to download Task Metadata Schema file. Please check internet connectivity and retry this action.')
+          end
+        end
+        JSON.parse(schema)
```

Comment:
Consider rescuing JSON parse error here (and logging a nicer error message) in case the schema is corrupted somehow